File: p_26.py
from decimal import Decimal, getcontext

getcontext().prec = 50


def calc(num, den):
    mp = {}
    rem = num % den
    res = ""
    while (rem != 0) and (rem not in mp.keys()):
        mp[rem] = len(res)
        rem *= 10
        res += str(rem // den)
        rem %= den

    return "-" if rem == 0 else res[mp[rem]:]


_min = 2
_max = 10

ans, l_ans = 0, 0

for i in range(_min, _max + 1):
    res = calc(1, i)
    if res != "-":
        print(i, res, len(res))
        if len(res) > l_ans:
            l_ans = len(res)
            ans = i

print("=======")
print(ans, l_ans)

print(1 / Decimal(183))

File: src/p_26.cpp
#include <bits/stdc++.h>

int main() {
    std::ios::sync_with_stdio(false);
    std::cin.tie(nullptr);
    int _min = 2, _max = 1000;
    auto Len = [=](int x, int y) -> int {
        std::map<int, int> mp;
        std::string res = "";
        int rem = x % y;
        while (rem != 0 && mp.find(rem) == mp.end()) {
            mp[rem] = res.size();
            rem *= 10;
            res += std::to_string(rem / y);
            rem %= y;
        }
        return rem == 0 ? -1 : res.size() - mp[rem];
    };
    int ans = 0, l_ans = 0;
    for (int i = _min; i <= _max; i++) {
        int len = Len(1, i);
        if (len != -1 && len > l_ans) {
            l_ans = len;
            ans = i;
        }
    }
    std::cout << ans << " => " << l_ans << "\n";
    return 0;
}